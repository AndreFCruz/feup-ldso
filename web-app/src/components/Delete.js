import React, {Component} from 'react';
import { Button } from 'reactstrap';
import PropTypes from 'prop-types';

class Delete extends Component{

    constructor(props){
        super(props);

        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleSubmit(event){
        event.preventDefault();

        const route = process.env.REACT_APP_ENDPOINT + 'faculties/en/' + this.props.match.params.faculty.toLowerCase() + '/social-projects?where[id]=' + this.props.match.params.project;
        fetch(route, {
            method: 'DELETE',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
            }
        });
    }

    render(){
        return(
            <Button color="secondary" onClick={this.handleSubmit}>Delete</Button>
        );
    }
}

Delete.propTypes = {
    match: PropTypes.object.isRequired
};


export default Delete;
