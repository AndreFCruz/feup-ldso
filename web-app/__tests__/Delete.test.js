import React from 'react';
import renderer from 'react-test-renderer';
import Delete from '../src/components/Delete';
import 'isomorphic-fetch'; //components use fetch() on mount
import MockAdapter from 'axios-mock-adapter';
import axios from 'axios';
import { shallow } from 'enzyme';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

var mockAxios = new MockAdapter(axios);

var props = {
    match: {
        params: {
            faculty: 'faup',
            project: '1'
        }
    }
};

describe('Delete Test Suite', () => {
    it('Deletes a project correctly', () => {

        mockAxios.onDelete('http://localhost:3000/faculties/en/faup/social-projects?where[id]=').reply(200, {
            id: 1
        });

        const mockupCall = jest.fn();
        const component = renderer.create(
            <Delete
                {...props}
                onChildSetRefresh={mockupCall}
            />
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Test click handle submit event', () => {

        const mockCallBack = jest.fn();

        const button = shallow((
            <Delete
                {...props}
                onChildSetRefresh={mockCallBack}
            />));
        button.find('Button').simulate('click',
            {
                preventDefault: () => { }
            });
        expect(mockCallBack.mock.calls.length).toEqual(1);

    });
});
