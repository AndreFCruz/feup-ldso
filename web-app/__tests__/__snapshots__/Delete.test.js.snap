// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete Test Suite Deletes a project correctly 1`] = `
<button
  aria-label={null}
  className="btn btn-secondary"
  onClick={[Function]}
  type="button"
>
  Delete
</button>
`;
